torch/csrc/autograd/python_function.cpp

```diff
@@ -1149,6 +1152,29 @@ int THPFunction_set_materialize_grads(
   END_HANDLE_TH_ERRORS_RET(-1)
 }
 
+PyObject* THPFunction_get_materialize_non_diff_grads(
+    THPFunction* self,
+    void* _unused) {
+  HANDLE_TH_ERRORS
+  return self->materialize_non_diff_grads ? Py_True : Py_False;
```

Comment:
You're missing refcounting!
Use the Py_RETURN_TRUE / Py_RETURN_FALSE macros

Reply:
good catch!